api/proto/metrics/metrics.proto

```diff
@@ -20,8 +20,40 @@ enum ZmetricTypes {
 message memoryMetric {
   uint32 usedMem = 2;            //in MBytes
   uint32 availMem = 3;           //in MBytes
-  double usedPercentage = 4;
-  double availPercentage = 5;
+  double usedPercentage = 4; // Obsolete - No Longer Used.
+  double availPercentage = 5; // Obsolete - No Longer Used.
+}
+
+// App Memory Metric
+message AppMemoryMetric {
+    // totalMB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 totalMB = 1;
+
+    // usedMB - Memory Used by app from totalMB Memory
+    //  usedMB <= totalMB in normal cases.
+    //  But usedMB can exceed totalMB - this could be a warning to the user.
+    uint32 usedMB = 2;
+}
+
+message DeviceMemoryMetric {
+  // totalMB - Total Device memory as seen by EVE.
```

Comment:
Please add that this is the same as resported in the memory field in the device info message repeated here for convenience.
Also, "as seen by EVE isn't informative and can be misleading (EVE only sees dom0 when running Xen etc)

Reply:
Can you call it "memory" i.e., the same as in the info API? That avoids using "total" which is a confusing term here. 
physicalMemory, memorySize, etc are alternative names.

Reply:
Done:

  // memoryMB - Total Device memory
  // This is the same as reported in the memory field in the device info
  // message repeated here for convenience.
  uint32 memoryMB = 1;